apps/astarte_realm_management/test/astarte_realm_management/device_removal/core_test.exs

```diff
@@ -0,0 +1,242 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Astarte.RealmManagement.DeviceRemover.CoreTest do
+  alias Astarte.DataAccess.Device.DeletionInProgress
+  alias Astarte.DataAccess.KvStore
+  alias Astarte.DataAccess.Groups.GroupedDevice
+  alias Astarte.DataAccess.Realms.Name
+  alias Astarte.DataAccess.Devices.Device
+  alias Astarte.Core.CQLUtils
+  alias Astarte.RealmManagement.Engine
+  alias Astarte.DataAccess.Realms.IndividualProperty
+  alias Astarte.RealmManagement.Migrations.CreateDatastreamIndividualMultiInterface
+  alias Astarte.RealmManagement.Queries
+  alias Astarte.DataAccess.Realms.IndividualDatastream
+  alias Astarte.DataAccess.Realms.Realm
+  alias Astarte.DataAccess.Repo
+  alias Astarte.RealmManagement.DeviceRemoval.Core
+
+  use Astarte.RealmManagement.DataCase, async: true
+  use ExUnitProperties
+
+  describe "Device remover Core" do
+    @describetag :device_remover
+
+    property "delete_individual_datastream/2 removes individual datastream data of a valid device",
+             %{realm: realm} do
+      keyspace = Realm.keyspace_name(realm)
+
+      Ecto.Migrator.run(Repo, [{0, CreateDatastreamIndividualMultiInterface}], :up,
+        prefix: keyspace,
+        all: true
+      )
+
+      check all(
+              interface <- Astarte.Core.Generators.Interface.interface(),
+              device <- Astarte.Core.Generators.Device.device(interfaces: interface),
+              endpoint_id = Ecto.UUID.bingenerate(),
+              value_timestamp = DateTime.utc_now(),
+              reception_timestamp = DateTime.utc_now(),
+              reception_timestamp_submillis <- integer(0..10)
+            ) do
+        %IndividualDatastream{
+          device_id: device.device_id,
+          interface_id: interface.interface_id,
+          endpoint_id: endpoint_id,
+          path: "/path/",
+          value_timestamp: value_timestamp,
+          reception_timestamp: reception_timestamp,
+          reception_timestamp_submillis: reception_timestamp_submillis
+        }
+        |> Repo.insert!(prefix: keyspace)
```

Comment:
As agreed upon IRL, it would be great to test this by seeding the table with ([list of](https://hexdocs.pm/stream_data/StreamData.html#list_of/1)) fully generated data, so that we're actually testing the property.